rl4co/envs/routing/tsp/env.py

```diff
@@ -167,5 +167,59 @@ def check_solution_validity(td: TensorDict, actions: torch.Tensor):
         ).all(), "Invalid tour"
 
     @staticmethod
-    def render(td: TensorDict, actions: torch.Tensor=None, ax = None):
+    def render(td: TensorDict, actions: torch.Tensor = None, ax=None):
         return render(td, actions, ax)
+
+
+# NOTE Experimental TSP class for stepwise PPO
+
+
+class TSPEnv4PPO(TSPEnv):
```

Comment:
[Minor] this may be called `DenseRewardTSPEnv` or similar?

Reply:
Sure! Btw., Stepwise PPO for TSP indeed converges to the nearest neighbor heuristic, at least with the stepwise reward as it is defined here (the distance added by the action):

<img width="513" alt="image" src="https://github.com/ai4co/rl4co/assets/66203600/d1d1e492-868d-4962-a871-1f8311f3e5f1">

Do you have a preferrence as to how to call the stepwise PPO in the paper (dense, stepwise something else)? And then we should probably adjust the description about PPO in the appendix